python/spider-py/src/spider_py/task_executor/task_executor.py

```diff
@@ -1,8 +1,182 @@
 """Executes a Spider Python task."""
 
+import argparse
+import inspect
+import logging
+from io import BufferedReader
+from os import fdopen
+from pydoc import locate
+from uuid import UUID
+
+import msgpack
+
+from spider_py import client
+from spider_py.storage import MariaDBStorage, parse_jdbc_url, Storage
+from spider_py.task_executor.task_executor_message import get_request_body, TaskExecutorResponseType
+from spider_py.utils import from_serializable, to_serializable
+
+# Set up logger
+logger = logging.getLogger(__name__)
+
+
+HeaderSize = 16
+
+
+def parse_args() -> argparse.Namespace:
+    """
+    Parses task executor arguments.
+    :return: The parsed arguments.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--func", type=str, required=True, help="Name of the function to execute.")
+    parser.add_argument(
+        "--libs", nargs="+", type=str, required=True, help="List of libraries to load."
+    )
+    parser.add_argument(
+        "--storage_url", type=str, required=True, help="JDBC URL for the storage backend."
+    )
+    parser.add_argument("--task_id", type=str, required=True, help="Task UUID.")
+    parser.add_argument(
+        "--input-pipe", type=int, required=True, help="File descriptor for the input pipe."
+    )
+    parser.add_argument(
+        "--output-pipe", type=int, required=True, help="File descriptor for the output pipe."
+    )
+    return parser.parse_args()
+
+
+def receive_message(pipe: BufferedReader) -> bytes:
+    """
+    Receives message from the pipe with a size header.
+    :param pipe: Pipe to receive message from.
+    :return: Received message body.
+    :raises EOFError: If the message body size does not match header size.
+    """
+    body_size_str = pipe.read(HeaderSize).decode()
+    body_size = int(body_size_str, base=10)
+    body = pipe.read(body_size)
+    if len(body) != body_size:
+        msg = "Received message body size does not match the header size."
+        raise EOFError(msg)
+    return body
+
+
+def parse_task_arguments(
+    storage: Storage, params: list[inspect.Parameter], arguments: list[object]
+) -> list[object]:
+    """
+    Parses arguments for the function to be executed.
+    Note that `params` does not include the `TaskContext` parameter, and must be the same length as
+    `arguments`.
```

Comment:
```suggestion

    NOTE: `params` does not include the `TaskContext` parameter, and must be the same length as
    `arguments`. The caller is responsible for the size check.

```
As indicated in the previous comment: https://github.com/y-scope/spider/pull/229#discussion_r2352580969
* `NOTE` is the notation we use for special alerts.
* As "explicitly add a NOTE to say the caller is responsible for the size checking", I don't think this comment makes a difference if it doesn't explicitly point out that the caller is responsible for the size checking.